### PR TITLE
Cleanup out defunct to do comments

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

The repo has a ton of `TODO` comments in it. I went through the ones in the test files and found most of them were defunct. (Or I just opened a ticket to cover them later.)

## Why is the change being made?

This is the beginning of me finally doing the work to purge ARMI of all `TODO` statements. I will see if they are still relevant. If they are, I will fix them or make them a ticket.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by john-science** [comment]
Not properly merged into main. I'll fix it.

File: armi/reactor/tests/test_blocks.py [hunk 1]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
     def test_rotatePins(self):
         b = self.block
         b.setRotationNum(0)
-        index = b._rotatePins(0, justCompute=True)
+        index = b.rotatePins(0, justCompute=True)
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1453]:
```suggestion
        index = b._rotatePins(0, justCompute=True)
```

File: armi/reactor/tests/test_blocks.py [hunk 2]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
         self.assertEqual(b.getRotationNum(), 0)
         self.assertEqual(index[5], 5)
         self.assertEqual(index[2], 2)  # pin 1 is center and never rotates.
 
-        index = b._rotatePins(1)
+        index = b.rotatePins(1)
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1458]:
```suggestion
        index = b._rotatePins(1)
```

File: armi/reactor/tests/test_blocks.py [hunk 3]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
         self.assertEqual(b.getRotationNum(), 1)
         self.assertEqual(index[2], 3)
         self.assertEqual(b.p.pinLocation[1], 3)
 
-        index = b._rotatePins(1)
+        index = b.rotatePins(1)
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1463]:
```suggestion
        index = b._rotatePins(1)
```

File: armi/reactor/tests/test_blocks.py [hunk 4]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
         self.assertEqual(b.getRotationNum(), 2)
         self.assertEqual(index[2], 4)
         self.assertEqual(b.p.pinLocation[1], 4)
 
-        index = b._rotatePins(2)
-        index = b._rotatePins(4)  # over-rotate to check modulus
+        index = b.rotatePins(2)
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1468]:
```suggestion
        index = b._rotatePins(2)
```

File: armi/reactor/tests/test_blocks.py [hunk 5]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
+        index = b.rotatePins(4)  # over-rotate to check modulus
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1469]:
```suggestion
        index = b._rotatePins(4)  # over-rotate to check modulus
```

File: armi/reactor/tests/test_blocks.py [hunk 6]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
         self.assertEqual(index[2], 4)
         self.assertEqual(index[6], 2)
         self.assertEqual(b.p.pinLocation[1], 4)
         self.assertEqual(b.p.pinLocation[5], 2)
 
-        self.assertRaises(ValueError, b._rotatePins, -1)
-        self.assertRaises(ValueError, b._rotatePins, 10)
-        self.assertRaises((ValueError, TypeError), b._rotatePins, None)
-        self.assertRaises((ValueError, TypeError), b._rotatePins, "a")
+        self.assertRaises(ValueError, b.rotatePins, -1)
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1476]:
```suggestion
        self.assertRaises(ValueError, b._rotatePins, -1)
```

File: armi/reactor/tests/test_blocks.py [hunk 7]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
+        self.assertRaises(ValueError, b.rotatePins, 10)
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1477]:
```suggestion
        self.assertRaises(ValueError, b._rotatePins, 10)
```

File: armi/reactor/tests/test_blocks.py [hunk 8]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
+        self.assertRaises((ValueError, TypeError), b.rotatePins, None)
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1478]:
```suggestion
        self.assertRaises((ValueError, TypeError), b._rotatePins, None)
```

File: armi/reactor/tests/test_blocks.py [hunk 9]
```diff
@@ -1454,33 +1450,33 @@ def test_106_getAreaFractions(self):
...
+        self.assertRaises((ValueError, TypeError), b.rotatePins, "a")
```
Comment (john-science) [inline armi/reactor/tests/test_blocks.py:1479]:
```suggestion
        self.assertRaises((ValueError, TypeError), b._rotatePins, "a")
```

**Review by opotowsky — APPROVED** [review]

**Pull request merged**